hive/src/main/java/com/netflix/iceberg/hive/HiveTableOperations.java

```diff
@@ -0,0 +1,224 @@
+    // if the metadata is already out of date, reject it
+    if (base != current()) {
+      throw new CommitFailedException("Cannot commit changes based on stale tableName metadata");
+    }
+
+    // if the metadata is not changed, return early
+    if (base == metadata) {
+      LOG.info("Nothing to commit.");
+      return;
+    }
+
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+
+    boolean threw = true;
+    Optional<Long> lockId = Optional.empty();
+    try {
+      lockId = acquireLock();
+      // TODO add lock heart beating for cases where default lock timeout is too low.
+      Table tbl;
+      if (base != null) {
+        tbl = metaStoreClient.getTable(database, tableName);
+      } else {
+        final long currentTimeMillis = System.currentTimeMillis();
+        final StorageDescriptor storageDescriptor = storageDescriptor(metadata.schema());
+        tbl = new Table(tableName,
+                database,
+                System.getProperty("user.name"),
+                (int) currentTimeMillis / 1000,
+                (int) currentTimeMillis / 1000,
+                Integer.MAX_VALUE,
+                storageDescriptor,
+                Collections.emptyList(),
+                new HashMap<>(),
+                null,
+                null,
+                ICEBERG_TABLE_TYPE_VALUE);
+      }
+
+      // unnecessary but setting in case tableName.getParameters() starts returning a deep copy.
+      setParameters(newMetadataLocation, tbl);
+
+      if (base != null) {
+        metaStoreClient.alter_table(database, tableName, tbl);
+      } else {
+        metaStoreClient.createTable(tbl);
+      }
+      threw = false;
+    } catch (TException|UnknownHostException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (threw) {
+        // if anything went wrong, clean up the uncommitted metadata file
+        deleteFile(newMetadataLocation);
+      }
+      unlock(lockId);
+    }
+
+    refresh();
+  }
+
+  private void setParameters(String newMetadataLocation, Table tbl) {
+    Map<String, String> parameters = tbl.getParameters();
+
+    if (parameters == null) {
+      parameters = new HashMap<>();
+    }
+    parameters.put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(METADATA_LOCATION_PROP, newMetadataLocation);
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+
+    tbl.setParameters(parameters);
+  }
+
+  private StorageDescriptor storageDescriptor(Schema schema) {
+    // TODO HardCoding the parquet for now
+    final StorageDescriptor storageDescriptor = new StorageDescriptor();
+    storageDescriptor.setCols(columns(schema));
+    storageDescriptor.setLocation(hiveTableLocation());
+    storageDescriptor.setOutputFormat("org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat");
```

Comment:
We updated the Metacat implementation to use org.apache.hadoop.mapred.FileInputFormat and  org.apache.hadoop.mapred.FileOutputFormat. Those can be loaded (so Hive can read the metadata) but can't be instantiated (they are abstract) so metadata is valid but the table can't be read unless you go through the Iceberg API.